cmake/CMakeLists.txt

```diff
@@ -1062,6 +1062,9 @@ if (onnxruntime_USE_CUDA)
     list(APPEND ONNXRUNTIME_CUDA_LIBRARIES cublas cudnn curand cufft)
   endif()
 
+  # Enable NCCL by default when using CUDA.
+  set(onnxruntime_USE_NCCL ON)
```

Comment:
NCCL is only used in ORT training now. Could you move this line to line #1195 below? It should be set when MPI is found.

Reply:
Done. When moving this line inside the MPI I enable NCCL only when CUDA is ON.